### PR TITLE
Used the memory-efficient representation of the GMFs

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1438212.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-risklib/498

### CODE REVIEW

File: openquake/commonlib/calculators/base.py [hunk 1]
```diff
@@ -153,7 +154,8 @@ def save_pik(self, result, **kw):
         :returns: a dictionary with the saved data
         """
         haz_out = dict(rlzs_assoc=self.rlzs_assoc,
-                       sitecol=self.sitecol, oqparam=self.oqparam)
+                       sites=getattr(self, 'sites', self.sitecol),
```
Comment (ptormene) [inline openquake/commonlib/calculators/base.py]: Please add a comment

File: openquake/commonlib/calculators/base.py [hunk 2]
```diff
@@ -332,3 +334,57 @@ def save_pik(self, result, **kw):
...
                 cPickle.dump(self.risk_out, f)
         return self.risk_out
+
+
+# functions useful for the calculators ScenarioDamage and ScenarioRisk
+
+
+def get_gmfs(self):
+    """
+    :param self: a ScenarioDamage or ScenarioRisk selfulator
```
Comment (ptormene) [inline openquake/commonlib/calculators/base.py]: Please check if the refactoring calc -> self produced any other words like selfulator

File: openquake/commonlib/calculators/base.py [hunk 3]
```diff
@@ -332,3 +334,57 @@ def save_pik(self, result, **kw):
             with open(cache, 'w') as f:
                 cPickle.dump(self.risk_out, f)
         return self.risk_out
+
+
+# functions useful for the calculators ScenarioDamage and ScenarioRisk
+
+
+def get_gmfs(self):
```
Comment (ptormene) [inline openquake/commonlib/calculators/base.py]: Here and in the following functions, please rename self -> calc (or other) and describe the param in the docstring

**Comment by ptormene** [comment]
Besides the comments, LGTM

File: openquake/commonlib/calculators/base.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/calculators/base.py]: Done

File: openquake/commonlib/calculators/base.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/base.py]: Done

**Pull request merged**